pontoon/sync/formats/__init__.py

```diff
@@ -37,68 +38,94 @@ def are_compatible_files(file_a, file_b):
     return False
 
 
-def as_vcs_translations(res: MozL10nResource[Message]) -> list[VCSTranslation]:
-    translations: list[VCSTranslation] = []
+_fluent_serializer = FluentSerializer()
+_android_esc_u = compile(r"(?<!\\)\\u[0-9A-Fa-f]{4}")
+_android_esc_char = compile(r"(?<!\\)\\([^nt])")
+_android_esc_nl = compile(r"(?<!\\)\\n\s*")
+_android_ws_around_outer_tag = compile(r"^\s+(?=<)|(?<=>)\s+$")
+_android_ws_before_block = compile(r"\s+(?=<(br|label|li|p|/?ul)\b)")
+_android_ws_after_block = compile(r"((?<=<br>)|(?<=<br/>)|(?<=</ul>)|(?<=\\n))\s+")
+_prop_esc_u = compile(r"(?<!\\)\\u(?!0000)[0-9A-Fa-f]{4}")
+_prop_esc_ws = compile(r"(?<!\\)\\([^\S\n])")
```

Comment:
This PR almost looks like an opportunity to move this to moz.l10n.

Reply:
These are actually rather Pontoon-special, and required because the serialization used in Pontoon doesn't _quite_ match up to what we write in repositories, especially for Android. Eventually, these should go away completely.